File: Algorithms/Implementation/Easy/Modified Kaprekar Numbers/python/kaprekar1/kaprekar.py
def kaprekarNumbers(p, q):
	kaprekar_numbers = list()

	for k in range(p, q + 1):
		d = len(str(k))
		square = str(pow(k, 2))
		l = int(square[: len(square) - d]) if len(square) > d else 0
		r = int(square[len(square) - d :])

		if l + r == k:
			kaprekar_numbers.append(k)

	return kaprekar_numbers


p = int(input())
q = int(input())
kaprekar_numbers = kaprekarNumbers(p, q)

if (len(kaprekar_numbers) > 0):
	for k in kaprekar_numbers:
		print(str(k) if k == kaprekar_numbers[-1] else str(k) + ' ', end='')
else:
	print("INVALID RANGE")

File: Algorithms/Implementation/Easy/Modified Kaprekar Numbers/javascript/kaprekar1/kaprekar.js
'use strict';

process.stdin.setEncoding('utf-8');
var os = require('os');
var input = "";

process.stdin.on("data", data => {
	input += data;

	if(data.split(os.EOL)[0] == "") {
		process.stdin.emit('end');
	}
});

process.stdin.on('end', _ => {
	main();
});

function main() {
	var limits = input.split(os.EOL).map(x=>+x);
	var kaprekar_numbers = kaprekar(limits);

	if (kaprekar_numbers.length > 0) {
		for (var k of kaprekar_numbers) {
			process.stdout.write(k + ' ');
		}
	} else {
		console.log("INVALID RANGE");
	}
}

function kaprekar(limits) {
	var kaprekar_numbers = [];

	for (var i = limits[0]; i <= limits[1]; i++) {
		let d = i.toString().length;
		let square = Math.pow(i, 2).toString();
		let l = parseInt(square.length > d ? square.substr(0, square.length - d) : 0);
		let r = parseInt(square.substr(square.length - d));

		if ((l + r) == i) {
			kaprekar_numbers.push(i);
		}
	}

	return kaprekar_numbers;
}